controller/test/sensor_tests/sensors_test.cpp

```diff
@@ -291,3 +291,55 @@ TEST(SensorTests, Calibration) {
                   .cubic_m_per_sec(),
               COMPARISON_TOLERANCE_FLOW_CUBIC_M_PER_SEC);
 }
+
+TEST(SensorTests, FlowDrift) {
+  // Set pressure sensors to 0kPa for calibration.
+  for (auto pin : {AnalogPin::PATIENT_PRESSURE, AnalogPin::INFLOW_PRESSURE_DIFF,
+                   AnalogPin::OUTFLOW_PRESSURE_DIFF}) {
+    Hal.test_setAnalogPin(pin, MPXV5004_PressureToVoltage(kPa(0)));
+  }
+
+  Sensors sensors;
+  sensors.Calibrate();
+
+  // Apply a constant pressure difference to the flow sensors over a few
+  // breaths.
+  std::vector<float> v1;
+  for (int i = 0; i < 10; i++) {
+    sensors.NoteNewBreath();
+    v1.push_back(
+        update_readings(seconds(6), cmH2O(0), cmH2O(0.005f), cmH2O(0), &sensors)
+            .volume_ml);
+  }
+  // At first TV should increase, but eventually this should be detected as
+  // flow sensor drift, and the TV should level off and start decreasing.
+  // Check that the max isn't the first or last element.
```

Comment:
Of course for now this is OK, but once we know what gains we are going to use for this, could we make it so those tests run more precise checks?

Reply:
Yes, absolutely!